schema/composite.gql

```diff
@@ -129,10 +140,15 @@ resource claim.
 """
 type CompositeResourceClaimSpec {
   """
-  The composition this composite resource uses to compose resources.
+  The composition this composite resource claim uses to compose resources.
   """
   composition: Composition @goField(forceResolver: true)
 
+  """
+  The compositionRef this composite resource claim uses to compose resources.
+  """
+  compositionRef: String
```

Comment:
Should this be an ObjectReference? That would more closely match the underlying "XRM" API.

Reply:
The existing code consuming this only pays attention to the `Name` part of the `ObjectReference`:
https://github.com/upbound/xgql/blob/e0ce8daa9a555b74d82f37c4373bcc1f66d752cf/internal/graph/resolvers/composite.go#L108-L131

I think this means that if you use anything other than the name it won't work. Should we fix that or assume that the ref is always just to a name?

Reply:
Yeah, we only need the name here. What I'm thinking though is that we should have the GraphQL schema here "mirror" the XRM API as closely as possible by "nesting" an object with only a single `name` field. So something like:

```graphql
type LocalObjectReference {
  name: String!
}

type CompositeResourceClaimSpec {
  compositionRef: LocalObjectReference
}
```

This is primarily a KRM API conventions thing - it's expected that when you see a field named `<t>Ref` its value will be a subset of the ObjectReference type rather than a string.

Reply:
Makes sense.